src/psyclone/nemo.py

```diff
@@ -58,15 +58,15 @@ class NemoFparser2Reader(Fparser2Reader):
     Specialisation of Fparser2Reader for the Nemo API.
     '''
     @staticmethod
-    def _create_schedule(_):
+    def _create_schedule(name):
         '''
         Create an empty InvokeSchedule. The un-named argument would be 'name'
         but this isn't used in the NEMO API.
 
         :returns: New InvokeSchedule empty object.
         :rtype: py:class:`psyclone.nemo.NemoInvokeSchedule`
         '''
-        return NemoInvokeSchedule()
+        return NemoInvokeSchedule(name)
 
     def _create_loop(self, parent, variable):
```

Comment:
I understand this can't be removed yet because the raising transformations are created but not used in the nemo path yet, but eventually it will replace this code. Is this right?

Looking at this method, is the assignment of a loop.loop_type given the variable name loop_type_mapping something that the CreateNemoLoopTrans should be doing? Or is it left out on purpose?

Reply:
Yes, that's right - it will be removed. Your second question made me think but in fact this assignment is now done in the `NemoLoop.create()` method. I can't use that here though because it requires that all the children of the loop be supplied.